pkg/config/config.go

```diff
@@ -86,9 +86,13 @@ func Get[T any](key string) (T, error) {
 		return zeroValue[T](), fmt.Errorf("value not found for %s", key)
 	}
 
+	var val T
 	val, ok := raw.(T)
 	if !ok {
-		return zeroValue[T](), fmt.Errorf("value not of expected type, got: %T", raw)
+		err := ForKey(key, &val)
+		if err != nil {
+			return zeroValue[T](), fmt.Errorf("value not of expected type, got: %T", raw)
+		}
```

Comment:
what is this change doing?

Reply:
If `Get` is used to get a custom struct then it will incorrectly return a `map[string]interface{}`. This change makes it correctly return a struct, so that you can do `Get[types.IpfsConfig](...)`

Reply:
🔥 love it.